nityam_kumar/PostgresAssignment/src/controller/mark.ts

```diff
@@ -0,0 +1,163 @@
+import { Context } from "vm";
+import * as marksWorker from "../services/mark";
+import { checkMark } from "../config/validation";
+
+import {
+  invalidData,
+  serverERROR,
+  NOTFOUNDERROR,
+  CONFLICTERROR,
+  UNAUTHORIZEDERROR,
+} from "../config/error";
+
+interface Mark {
+  st_id: string;
+  sub_id: string;
+  marks: number;
+  teacher_id: string;
+  cl_no: number;
+}
+
+export const createMarks = async (ctx: Context) => {
+  try {
+    const m1: Mark = ctx.request.body;
+    if (!checkMark(m1)) {
+      invalidData(ctx, "bad input data");
+      return;
+    }
+
+    if (!(await marksWorker.checkStudentExist(m1))) {
```

Comment:
we can throw error from all these methods and handle it in catch block, check with druval on this.